apps/core/src/hooks/useTransactionSummary.ts

```diff
@@ -45,6 +50,140 @@ const getSummary = (
     }
 };
 
+const getObjectSummaryNFTData = <T extends { objectId: string }>({
+    nftsMeta,
+    originalData,
+}: {
+    nftsMeta: ReturnType<typeof useNFTsMeta>;
+    originalData: T[];
```

Comment:
This generic feels really weird, why do we need a generic vs just defining the actual types here? We can also force consumers to map into useful data constructs too